dashboards-observability/public/components/event_analytics/explorer/explorer.tsx

```diff
@@ -678,10 +684,13 @@ export const Explorer = ({
                           <TimechartHeader
                             dateFormat={'MMM D, YYYY @ HH:mm:ss.SSS'}
                             options={timeIntervalOptions}
-                            onChangeInterval={(intrv) => {
+                            onChangeInterval={(selectedIntrv) => {
+                              const intervalOptionsIndex = timeIntervalOptions.findIndex(item => item.value === selectedIntrv)
+                              const intrv = selectedIntrv.length > 2 ? selectedIntrv.slice(5) : selectedIntrv
                               getCountVisualizations(intrv);
+                              selectedIntervalRef.current = timeIntervalOptions[intervalOptionsIndex]
                             }}
-                            stateInterval="auto"
+                            stateInterval={selectedIntervalRef.current?.value || ''}
```

Comment:
when would `selectedIntervalRef.current?.value` not exist?

Reply:
I was thinking it may not exist before we have a chance to call fetch data - but maybe I am overthinking - should I just use a `!` and assert that it exists by the time we render the timechart header?

Reply:
i think you can change https://github.com/opensearch-project/observability/blob/af2c6acbf860b5246cbe0a92dc7f9112cdd81377/dashboards-observability/public/components/event_analytics/explorer/timechart_header/timechart_header.tsx#L35
to `stateInterval?: string` and remove the `|| ''` here

Reply:
Fixed in [25ed4ec](https://github.com/opensearch-project/observability/pull/1266/commits/25ed4ec02ba006512631e22aacf038f9ca599525)